cpp/ql/test/library-tests/ir/ir/ir.cpp

```diff
@@ -1219,4 +1219,19 @@ void switch2Case_default(int x) {
     int z = y;
 }
 
+int staticLocalInit(int x) {
+    static int a = 0;  // Constant initialization
+    static int b = sizeof(x);  // Constant initialization
+    static int c = x;  // Dynamic initialization
+    static int d;  // Zero initialization
+
+    return a + b + c + d;
+}
+
+void staticLocalWithConstructor(const char* dynamic) {
+    static String a;
+    static String b("static");
+    static String c(dynamic);
```

Comment:
These three inits are all classified as dynamic. Are they supposed to be? Do the constructors for `a` and `b` run before `main` or upon first entering this function?

Reply:
Each initializer runs upon the first execution of the declaration statement of the corresponding variable, just like any other initializer that contains a (non-`constexpr`) function call.